src/sentry/integrations/slack/tasks.py

```diff
@@ -120,3 +126,46 @@ def find_channel_id_for_rule(project, actions, uuid, rule_id=None, **kwargs):
         return
     # if we never find the channel name we failed :(
     redis_rule_status.set_value("failed")
+
+
+@instrumented_task(
+    name="sentry.integrations.slack.search_channel_id_metric_alerts", queue="integrations"
+)
+def find_channel_id_for_alert_rule(organization_id, uuid, data, alert_rule_id=None):
+    redis_rule_status = RedisRuleStatus(uuid)
+    try:
+        organization = Organization.objects.get(id=organization_id)
+    except Organization.DoesNotExist:
+        redis_rule_status.set_value("failed")
+        return
+
+    alert_rule = None
+    if alert_rule_id:
+        try:
+            alert_rule = AlertRule.objects.get(organization_id=organization_id, id=alert_rule_id)
+        except AlertRule.DoesNotExist:
+            redis_rule_status.set_value("failed")
+            return
+
+    # we use SystemAccess here because we can't pass the access instance from the request into the task
+    # this means at this point we won't raise any validation errors associated with permissions
+    # however, we should only be calling this task after we tried saving the alert rule first
+    # which will catch those kinds of validation errors
+    serializer = AlertRuleSerializer(
+        context={"organization": organization, "access": SystemAccess(), "use_async_lookup": True},
+        data=data,
+        instance=alert_rule,
+    )
+    if serializer.is_valid():
+        try:
+            alert_rule = serializer.save()
+            redis_rule_status.set_value("success", alert_rule.id)
+            return
+        # we can still get a validation error for the channel not existing
+        except (serializers.ValidationError, ChannelLookupTimeoutError):
+            # channel doesn't exist error or validation error
+            redis_rule_status.set_value("failed")
```

Comment:
Should we pass some extra context back here about why we failed?

Reply:
@wedamija We probably should but the Redis cache we are using doesn't actually store that. If we do that, we'd need to update the async lookup for issue alerts. Most of the errors here will be that the channel doesn't exist or we don't have the perms so we _usually_ tell the user the right error message anyways lol.